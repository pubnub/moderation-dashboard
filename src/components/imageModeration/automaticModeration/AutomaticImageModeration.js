import React from "react";
import {
  Grid,
  Button,
  Box,
  Card,
  Chip,
  Typography,
  TextField,
  Select,
  Checkbox,
  CircularProgress,
  FormControl,
  MenuItem,
} from "@material-ui/core";
import { useStyles } from "../../../style/imageModeration";
import Slider from "../../core/SliderComponent";
import { LightTooltip } from "../../../style/tooltip";
import { handleChange } from "./ImageModerationHandler";
import { constantBoolean } from "../../../utils/helpers";

const TooltipText = (
  <>
    <span>Please follow steps listed </span>
    <a
      style={{ fontWeight: 600, color: "black", textDecorationLine: "none" }}
      href="https://sightengine.com/docs/image-moderation-workflows"
    >
      here
    </a>
    <span> and input values</span>
  </>
);

const AutomaticImageModeration = ({ state, setState, handleSave }) => {
  const classes = useStyles();
  const checkForApplyToAllChannelIds = constantBoolean(state.applyToAllChannelIds);
  const checkboxForReroute = constantBoolean(state.reRouteMessages);

  if (!state.initialLoading) {
    return (
      <>
        <Card className={classes.card}>
          <Box>
            <Grid container>
              <Grid item>
                <label testid="Channel_ID" className={classes.labelText}>
                  Channel ID
                </label>
              </Grid>
              <Grid item>
                <Box className={classes.infoIcon}>
                  <LightTooltip title="Note: Please add a Channel ID or a Channel pattern. eg. channel.* OR pubNub. Channel ID pattern applies to BOTH text & image moderation if both are enabled.">
                    <img
                      src={process.env.PUBLIC_URL + "/images/info-circle.svg"}
                      alt="info-circle"
                    />
                  </LightTooltip>
                </Box>
              </Grid>
            </Grid>

            <Box pt={1} mb={2}>
              <TextField
                id="channelId"
                variant="outlined"
                name="channelId"
                error={state.channelIdError}
                disabled={checkForApplyToAllChannelIds || state.channelIdDisabled}
                placeholder="channel ID"
                value={state.channelId}
                onChange={handleChange({ setState, state })}
                size="small"
                fullWidth
              />
            </Box>
          </Box>
          <Grid item>
            <Grid container justify="flex-start">
              <Grid item>
                <Checkbox
                  id="checkBox"
                  checked={checkForApplyToAllChannelIds}
                  onChange={handleChange({
                    setState,
                    state,
                    name: "applyToAllChannelIds",
                  })}
                  disabled={state.channelIdDisabled}
                />
              </Grid>
              <Grid item>
                <Box pt={1} pl={1}>
                  <Typography testid="Apply_All" variant="body1">
                    Apply to All Channel IDs
                  </Typography>
                </Box>
              </Grid>
            </Grid>
          </Grid>
          <br />
          <Grid container justify="flex-start">
            <Grid item sm={12}>
              <label id="SelectTool" className={classes.labelText}>
                Select third party tool for Automatic Detection
              </label>
              <Box pt={1} mb={2}>
                <FormControl size="small" fullWidth>
                  <Select
                    id="ImageModerationTool"
                    variant="outlined"
                    value={state.toolForImageModeration}
                    name="toolForImageModeration"
                    onChange={handleChange({ setState, state })}
                    MenuProps={{
                      anchorOrigin: {
                        vertical: "bottom",
                        horizontal: "left",
                      },
                      getContentAnchorEl: null,
                    }}
                  >
                    <MenuItem value="sightengine">SightEngine</MenuItem>
                  </Select>
                </FormControl>
              </Box>
            </Grid>
          </Grid>
          <br />
          <Box pt={1}>
            <Grid container>
              <Grid item>
                <label testId="ApiUser" className={classes.labelText}>
                  Sightengine API User
                </label>
              </Grid>
              <Grid item>
                <Box className={classes.infoIcon}>
                  <LightTooltip title={TooltipText} interactive>
                    <img
                      src={process.env.PUBLIC_URL + "/images/info-circle.svg"}
                      alt="info-circle"
                    />
                  </LightTooltip>
                </Box>
              </Grid>
            </Grid>
            <Box pt={1} mb={2}>
              <TextField
                id="sightEngineAPIUser"
                variant="outlined"
                name="sightengineAPIUser"
                error={state.sightengineAPIUserError}
                helperText={state.sightengineAPIUserError && "Sightengine API User is required"}
                value={state.sightengineAPIUser}
                onChange={handleChange({ setState, state })}
                size="small"
                placeholder="Sightengine API User"
                fullWidth
              />
            </Box>
          </Box>
          <br />
          <Box pt={1}>
            <Grid container>
              <Grid item>
                <label testId="Api_Key" className={classes.labelText}>
                  Sightengine API Key
                </label>
              </Grid>
              <Grid item>
                <Box className={classes.infoIcon}>
                  <LightTooltip title={TooltipText} interactive>
                    <img
                      src={process.env.PUBLIC_URL + "/images/info-circle.svg"}
                      alt="info-circle"
                    />
                  </LightTooltip>
                </Box>
              </Grid>
            </Grid>
            <Box pt={1} mb={2}>
              <TextField
                id="apiKey"
                variant="outlined"
                name="sightengineAPIKey"
                error={state.sightengineAPIKeyError}
                helperText={state.sightengineAPIKeyError && "Sightengine API Key is required"}
                value={state.sightengineAPIKey}
                onChange={handleChange({ setState, state })}
                size="small"
                fullWidth
                placeholder="Sightengine API Key"
              />
            </Box>
          </Box>
          <br />
          <Box pt={1}>
            <Grid container>
              <Grid item>
                <label testId="workFlowId" className={classes.labelText}>
                  Sightengine Workflow ID
                </label>
              </Grid>
              <Grid item>
                <Box className={classes.infoIcon}>
                  <LightTooltip title={TooltipText} interactive>
                    <img
                      src={process.env.PUBLIC_URL + "/images/info-circle.svg"}
                      alt="info-circle"
                    />
                  </LightTooltip>
                </Box>
              </Grid>
            </Grid>

            <Box pt={1} mb={2}>
              <TextField
                id="sightEngineWorkflowId"
                name="sightengineWorkflowId"
                variant="outlined"
                size="small"
                error={state.sightengineWorkflowIdError}
                helperText={
                  state.sightengineWorkflowIdError && "Sightengine Workflow ID is required"
                }
                value={state.sightengineWorkflowId}
                onChange={handleChange({ setState, state })}
                fullWidth
                placeholder="Sightengine Workflow ID"
              />
            </Box>
          </Box>
          <br />
          <Grid container justify="space-between" spacing={6}>
            <Grid item sm={6} xs={12} md={6} lg={6}>
              <Box>
                <label testId="riskFactor" className={classes.labelText}>
                  Sightengine Risk Factor
                </label>
                <Box pt={1}>
                  <Slider
                    id="riskFactorSlider"
                    value={state.sightengineRiskFactorThreshold}
                    onChange={handleChange({
                      setState,
                      state,
                      name: "sightengineRiskFactorThreshold",
                    })}
                  />
                </Box>
              </Box>
            </Grid>
          </Grid>
          <br />
          <Grid container>
            <Grid item>
              <Grid container justify="flex-start">
                <Grid item>
                  <Checkbox
                    id="reRouteCheckBox"
                    name="reRouteMessages"
                    checked={checkboxForReroute}
                    onChange={handleChange({ setState, state })}
                  />
                </Grid>
                <Grid item>
                  <Box pt={1} pl={1}>
                    <Typography testid="reRouteMsg" variant="body1">
                      Route messages to
                    </Typography>
                  </Box>
                </Grid>
              </Grid>
            </Grid>
            <Grid item>
              <Box>
                <Chip className={classes.reroute} label={"banned." + state.channelId} />
              </Box>
            </Grid>
          </Grid>
          <br />
          <Grid container justify="space-between" spacing={3}>
            <Grid item></Grid>
            <Grid item>
              <Button
                id="save"
                className={classes.save}
                disabled={state.saveLoading}
                onClick={handleSave}
                variant="contained"
              >
                {state.saveLoading ? (
                  <CircularProgress
                    size={25}
                    thickness={4}
                    color="primary"
                    className={classes.loader}
                  />
                ) : (
                  <></>
                )}
                Save
              </Button>
            </Grid>
          </Grid>
        </Card>
      </>
    );
  } else {
    return <></>;
  }
};

export default AutomaticImageModeration;
