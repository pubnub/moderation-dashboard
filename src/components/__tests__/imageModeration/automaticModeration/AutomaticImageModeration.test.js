import React from "react";
import { shallow } from "enzyme";
import AutomaticImageModeration from "../../../imageModeration/automaticModeration/AutomaticImageModeration";

const mockState = {
  applyToAllChannelIds: true,
  initialLoading: false,
  imageModerationToggle: "",
  channelId: "initialText",
  channelIdError: false,
  sightengineAPIUserError: false,
  sightengineAPIKeyError: false,
  sightengineWorkflowIdError: false,
  channelOnChange: false,
  toolForImageModeration: "sightengine",
  sightengineAPIUser: "",
  sightengineAPIKey: "",
  sightengineWorkflowId: "",
  sightengineRiskFactorThreshold: "",
  reRouteMessages: "",
  error: {
    status: false,
    msg: "",
  },
  successMsg: "",
  errorMsg: "",
  successStatus: false,
  errorStatus: false,
  saveLoading: false,
};

const expectValue = (a, b) => {
  expect(a).toBe(b);
};

describe("Automatic Image Moderation Test Cases", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<AutomaticImageModeration state={mockState} setState={jest.fn()} />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Contains Labels", () => {
    expectValue(wrapper.find("[testid='Channel_ID']").text(), "Channel ID");
    expectValue(wrapper.find('[testid="Apply_All"]').text(), "Apply to All Channel IDs");
    expectValue(
      wrapper.find("#SelectTool").text(),
      "Select third party tool for Automatic Detection"
    );
    expectValue(wrapper.find('[testId="ApiUser"]').text(), "Sightengine API User");
    expectValue(wrapper.find('[testId="Api_Key"]').text(), "Sightengine API Key");
    expectValue(wrapper.find('[testId="workFlowId"]').text(), "Sightengine Workflow ID");
    expectValue(wrapper.find('[testId="riskFactor"]').text(), "Sightengine Risk Factor");
    expectValue(wrapper.find('[testid="reRouteMsg"]').text(), "Route messages to");
  });

  test("Check initial Values", () => {
    expect(wrapper.find("#channelId").getElement().props.value).toBe("initialText");
    expect(wrapper.find("#ImageModerationTool").getElement().props.value).toBe("sightengine");
    expect(wrapper.find("#sightEngineAPIUser").getElement().props.value).toBe("");
    expect(wrapper.find("#apiKey").getElement().props.value).toBe("");
    expect(wrapper.find("#sightEngineWorkflowId").getElement().props.value).toBe("");
    expect(wrapper.find("#riskFactorSlider").getElement().props.value).toBe("");

    expect(wrapper.find("#channelId").getElement().props.placeholder).toBe("channel ID");
    expect(wrapper.find("#sightEngineAPIUser").getElement().props.placeholder).toBe(
      "Sightengine API User"
    );

    expect(wrapper.find("#apiKey").getElement().props.placeholder).toBe("Sightengine API Key");
    expect(wrapper.find("#sightEngineWorkflowId").getElement().props.placeholder).toBe(
      "Sightengine Workflow ID"
    );
  });

  test("Check CheckBoxes", () => {
    expect(wrapper.find("#checkBox").getElement().props.checked).toBe(true);
    expect(wrapper.find("#reRouteCheckBox").getElement().props.checked).toBe(false);
  });

  test("Check Save Button", () => {
    expect(wrapper.find("#save").text()).toBe("Save");
  });

  test("SightEngine Handler", () => {
    const ChannelChange = wrapper.find("#channelId").simulate("change", {
      target: { name: "channelId", value: "newChannelId" },
    });
    expect(ChannelChange).toBeTruthy();

    const ToolChange = wrapper.find("#ImageModerationTool").simulate("change", {
      target: { name: "toolForImageModeration", value: "newSightEngine" },
    });
    expect(ToolChange).toBeTruthy();

    const ApiUserChange = wrapper.find("#sightEngineAPIUser").simulate("change", {
      target: { name: "sightengineAPIUser", value: "newApiUser" },
    });
    expect(ApiUserChange).toBeTruthy();

    const ApiKeyChange = wrapper.find("#apiKey").simulate("change", {
      target: { name: "sightengineAPIKey", value: "newApiKey" },
    });
    expect(ApiKeyChange).toBeTruthy();

    const WorkFlowIdChange = wrapper.find("#sightEngineWorkflowId").simulate("change", {
      target: { name: "sightengineWorkflowId", value: "newWorkId" },
    });
    expect(WorkFlowIdChange).toBeTruthy();

    const ReRouteChange = wrapper.find("#reRouteCheckBox").simulate("change", {
      target: { name: "reRouteMessages", checked: true },
    });
    expect(ReRouteChange).toBeTruthy();

    const RiskChange = wrapper
      .find("#riskFactorSlider")
      .simulate("change", { target: { value: "" } }, "newRisk");
    expect(RiskChange).toBeTruthy();

    const ApplyAllTrue = wrapper.find("#checkBox").simulate("change", {
      target: { name: "applyToAllChannelIds", checked: true },
    });
    expect(ApplyAllTrue).toBeTruthy();

    const ApplyAllFalse = wrapper.find("#checkBox").simulate("change", {
      target: { name: "applyToAllChannelIds", checked: false },
    });
    expect(ApplyAllFalse).toBeTruthy();
  });
});
