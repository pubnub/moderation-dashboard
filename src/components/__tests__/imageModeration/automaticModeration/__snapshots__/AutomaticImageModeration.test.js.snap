// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Automatic Image Moderation Test Cases Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-2"
  >
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <label
            className="makeStyles-labelText-1"
            testid="Channel_ID"
          >
            Channel ID
          </label>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            className="makeStyles-infoIcon-9"
          >
            <WithStyles(WithStyles(ForwardRef(Tooltip)))
              title="Note: Please add a Channel ID or a Channel pattern. eg. channel.* OR pubNub. Channel ID pattern applies to BOTH text & image moderation if both are enabled."
            >
              <img
                alt="info-circle"
                src="/images/info-circle.svg"
              />
            </WithStyles(WithStyles(ForwardRef(Tooltip)))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(MuiBox)
        mb={2}
        pt={1}
      >
        <WithStyles(ForwardRef(TextField))
          disabled={true}
          error={false}
          fullWidth={true}
          id="channelId"
          name="channelId"
          onChange={[Function]}
          placeholder="channel ID"
          size="small"
          value="initialText"
          variant="outlined"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-start"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Checkbox))
            checked={true}
            id="checkBox"
            onChange={[Function]}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            pl={1}
            pt={1}
          >
            <WithStyles(ForwardRef(Typography))
              testid="Apply_All"
              variant="body1"
            >
              Apply to All Channel IDs
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={12}
      >
        <label
          className="makeStyles-labelText-1"
          id="SelectTool"
        >
          Select third party tool for Automatic Detection
        </label>
        <Styled(MuiBox)
          mb={2}
          pt={1}
        >
          <WithStyles(ForwardRef(FormControl))
            fullWidth={true}
            size="small"
          >
            <WithStyles(ForwardRef(Select))
              MenuProps={
                Object {
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "getContentAnchorEl": null,
                }
              }
              id="ImageModerationTool"
              name="toolForImageModeration"
              onChange={[Function]}
              value="sightengine"
              variant="outlined"
            >
              <WithStyles(ForwardRef(MenuItem))
                value="sightengine"
              >
                SightEngine
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <Styled(MuiBox)
      pt={1}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <label
            className="makeStyles-labelText-1"
            testId="ApiUser"
          >
            Sightengine API User
          </label>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            className="makeStyles-infoIcon-9"
          >
            <WithStyles(WithStyles(ForwardRef(Tooltip)))
              interactive={true}
              title={
                <React.Fragment>
                  <span>
                    Please follow steps listed 
                  </span>
                  <a
                    href="https://sightengine.com/docs/image-moderation-workflows"
                    style={
                      Object {
                        "color": "black",
                        "fontWeight": 600,
                        "textDecorationLine": "none",
                      }
                    }
                  >
                    here
                  </a>
                  <span>
                     and input values
                  </span>
                </React.Fragment>
              }
            >
              <img
                alt="info-circle"
                src="/images/info-circle.svg"
              />
            </WithStyles(WithStyles(ForwardRef(Tooltip)))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(MuiBox)
        mb={2}
        pt={1}
      >
        <WithStyles(ForwardRef(TextField))
          error={false}
          fullWidth={true}
          helperText={false}
          id="sightEngineAPIUser"
          name="sightengineAPIUser"
          onChange={[Function]}
          placeholder="Sightengine API User"
          size="small"
          value=""
          variant="outlined"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <br />
    <Styled(MuiBox)
      pt={1}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <label
            className="makeStyles-labelText-1"
            testId="Api_Key"
          >
            Sightengine API Key
          </label>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            className="makeStyles-infoIcon-9"
          >
            <WithStyles(WithStyles(ForwardRef(Tooltip)))
              interactive={true}
              title={
                <React.Fragment>
                  <span>
                    Please follow steps listed 
                  </span>
                  <a
                    href="https://sightengine.com/docs/image-moderation-workflows"
                    style={
                      Object {
                        "color": "black",
                        "fontWeight": 600,
                        "textDecorationLine": "none",
                      }
                    }
                  >
                    here
                  </a>
                  <span>
                     and input values
                  </span>
                </React.Fragment>
              }
            >
              <img
                alt="info-circle"
                src="/images/info-circle.svg"
              />
            </WithStyles(WithStyles(ForwardRef(Tooltip)))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(MuiBox)
        mb={2}
        pt={1}
      >
        <WithStyles(ForwardRef(TextField))
          error={false}
          fullWidth={true}
          helperText={false}
          id="apiKey"
          name="sightengineAPIKey"
          onChange={[Function]}
          placeholder="Sightengine API Key"
          size="small"
          value=""
          variant="outlined"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <br />
    <Styled(MuiBox)
      pt={1}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <label
            className="makeStyles-labelText-1"
            testId="workFlowId"
          >
            Sightengine Workflow ID
          </label>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            className="makeStyles-infoIcon-9"
          >
            <WithStyles(WithStyles(ForwardRef(Tooltip)))
              interactive={true}
              title={
                <React.Fragment>
                  <span>
                    Please follow steps listed 
                  </span>
                  <a
                    href="https://sightengine.com/docs/image-moderation-workflows"
                    style={
                      Object {
                        "color": "black",
                        "fontWeight": 600,
                        "textDecorationLine": "none",
                      }
                    }
                  >
                    here
                  </a>
                  <span>
                     and input values
                  </span>
                </React.Fragment>
              }
            >
              <img
                alt="info-circle"
                src="/images/info-circle.svg"
              />
            </WithStyles(WithStyles(ForwardRef(Tooltip)))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(MuiBox)
        mb={2}
        pt={1}
      >
        <WithStyles(ForwardRef(TextField))
          error={false}
          fullWidth={true}
          helperText={false}
          id="sightEngineWorkflowId"
          name="sightengineWorkflowId"
          onChange={[Function]}
          placeholder="Sightengine Workflow ID"
          size="small"
          value=""
          variant="outlined"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="space-between"
      spacing={6}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={6}
        md={6}
        sm={6}
        xs={12}
      >
        <Styled(MuiBox)>
          <label
            className="makeStyles-labelText-1"
            testId="riskFactor"
          >
            Sightengine Risk Factor
          </label>
          <Styled(MuiBox)
            pt={1}
          >
            <SliderComponent
              id="riskFactorSlider"
              onChange={[Function]}
              value=""
            />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          justify="flex-start"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
              id="reRouteCheckBox"
              name="reRouteMessages"
              onChange={[Function]}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              pl={1}
              pt={1}
            >
              <WithStyles(ForwardRef(Typography))
                testid="reRouteMsg"
                variant="body1"
              >
                Route messages to
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)>
          <WithStyles(ForwardRef(Chip))
            className="makeStyles-reroute-8"
            label="banned.initialText"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="space-between"
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-save-4"
          disabled={false}
          id="save"
          variant="contained"
        >
          Save
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Card))>
</Fragment>
`;
